audit.js

```diff
@@ -3,54 +3,61 @@ const assert = require('assert')
 const os = require('os')
 const uuidv4 = require('uuid/v4')
 
-module.exports = function (context, opts) {
-  assert(opts, 'Options object must be defined')
-  assert(typeof opts.eventName === 'string', 'Event name must be a string')
-  assert(typeof opts.resource === 'string', 'Resource must be a string')
-
-  context['audit-options'] = opts
-  context['audit-queue'] = []
-  context['audit-system'] = `${os.hostname()}-${os.platform()}`
-
-  context['audit'] = extensionLogArgs => {
-    const { eventName, resource } = context['audit-options']
-
-    context['audit-queue'].push({
-      auditId: uuidv4(),
-      eventName: eventName,
-      system: context['audit-system'],
-      date: new Date().toISOString(),
-      resource: resource,
-      extensionLog: Array.from(extensionLogArgs)
-    })
-  }
-
-  shimmer.wrap(context, 'log', function (original) {
-    return function () {
-      context['audit'](arguments)
-      return original.apply(this, arguments)
-    }
+const auditSystem = `${os.hostname()}-${os.platform()}`
+let auditQueue = []
+
+const audit = (extensionLogArgs, auditOptions) => {
+  const { eventName, resource } = auditOptions
+
+  auditQueue.push({
+    auditId: uuidv4(),
+    eventName: eventName,
+    system: auditSystem,
+    date: new Date().toISOString(),
+    resource: resource,
+    extensionLog: typeof extensionLogArgs === 'string' ? [ extensionLogArgs ] : Array.from(extensionLogArgs)
   })
+}
+
+module.exports = {
```

Comment:
i don't think we should actually call this `auditDirectly`, that was an example sorry for any confusion! maybe `audit` or similar depending on the verbiage expected when a dev references this module